src/Datasource/EntityTrait.php

```diff
@@ -748,6 +781,59 @@ public function extractOriginalChanged(array $fields): array
         return $result;
     }
 
+    /**
+     * Returns whether a field is an original one
+     *
+     * @return bool
+     */
+    public function isOriginalField(string $name): bool
+    {
+        return in_array($name, $this->_originalFields);
+    }
+
+    /**
+     * Returns an array of field names previously set using `setOriginalField()`
+     * The entity was initialized with
+     *
+     * @return array<string>
+     */
+    public function getOriginalFields(): array
+    {
+        return $this->_originalFields;
+    }
+
+    /**
+     * Sets the given field or a list of fields to as original.
+     * Normally there is no need to call this method manually.
+     *
+     * @param array<string>|string $field the name of a field or a list of fields to set as original
+     * @param bool $merge
+     * @return $this
+     */
+    public function setOriginalField(string|array $field, bool $merge = true)
+    {
+        if (!$merge) {
+            $this->_originalFields = (array)$field;
+
+            //WIP? Tests with assertEqual fail as long as the values of $this->_originalFields aren't in the same order.
+            sort($this->_originalFields);
```

Comment:
sort() shouldnt be needed
assertEquals() will always work with arrays and irrelevant order, whereas assertSame() would require exact order (which we cannot and dont want to guarantee here anyway).